src/lib.rs

```diff
@@ -931,6 +933,28 @@ macro_rules! __impl_bitflags {
     };
 }
 
+// When "arbitrary" is not enabled, emit code to implement the `Arbitrary` trait.
+#[cfg(feature = "impl_arbitrary")]
+#[macro_export(local_inner_macros)]
+#[doc(hidden)]
+macro_rules! __impl_arbitrary_for_bitflags {
+    ($BitFlags:ident) => {
+        impl<'a> arbitrary::Arbitrary<'a> for $BitFlags {
+            fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+                 Self::from_bits(u.arbitrary()?).ok_or_else(|| arbitrary::Error::IncorrectFormat)
+            }
+        }
+    };
+}
+
+// When "arbitrary" is not enabled, don't emit any code for it.
+#[cfg(not(feature = "impl_arbitrary"))]
+#[macro_export(local_inner_macros)]
+#[doc(hidden)]
+macro_rules! __impl_arbitrary_for_bitflags {
+    ($BitFlags:ident) => {};
+}
```

Comment:
```suggestion
// When "impl_arbitrary" is enabled, emit code to implement the `Arbitrary` trait.
#[cfg(feature = "impl_arbitrary")]
#[macro_export]
#[doc(hidden)]
macro_rules! __impl_arbitrary_for_bitflags {
    ($BitFlags:ident) => {
        impl<'a> arbitrary::Arbitrary<'a> for $BitFlags {
            fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
                 Self::from_bits(u.arbitrary()?).ok_or_else(|| arbitrary::Error::IncorrectFormat)
            }
        }
    };
}

// When "impl_arbitrary" is not enabled, don't emit any code for it.
#[cfg(not(feature = "impl_arbitrary"))]
#[macro_export]
#[doc(hidden)]
macro_rules! __impl_arbitrary_for_bitflags {
    ($BitFlags:ident) => {};
}
```
I'd hope there's a less ugly way to do this, but I don't think there is, since the flag needs to be evaluated here and not at the expansion.